From bbcda8a92ac9d692cc864eb99e2d19c23736b300 Mon Sep 17 00:00:00 2001
From: Bing Jiang <bing.jiang@amlogic.com>
Date: Mon, 2 Nov 2020 16:02:04 +0800
Subject: [PATCH] uboot: add set root system [1/1]

PD#SWPL-35835

Problem:
  add set root system

Solution:
  add set root system

Verify:
  SC2_AH212

Change-Id: I71d09b8a425f151aba55ea83b46a19540104dfbd
Signed-off-by: Bing Jiang <bing.jiang@amlogic.com>
---
 board/amlogic/configs/sc2_ah212.h            | 9 +++++++++
 board/amlogic/defconfigs/sc2_ah212_defconfig | 2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/board/amlogic/configs/sc2_ah212.h b/board/amlogic/configs/sc2_ah212.h
index 3885365363..081a3e45ef 100644
--- a/board/amlogic/configs/sc2_ah212.h
+++ b/board/amlogic/configs/sc2_ah212.h
@@ -187,6 +187,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
diff --git a/board/amlogic/defconfigs/sc2_ah212_defconfig b/board/amlogic/defconfigs/sc2_ah212_defconfig
index 026f88eb89..39a5570125 100644
--- a/board/amlogic/defconfigs/sc2_ah212_defconfig
+++ b/board/amlogic/defconfigs/sc2_ah212_defconfig
@@ -36,7 +36,7 @@ CONFIG_CMD_SARADC=y
 CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 # CONFIG_CMD_RSVMEM is not set
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
-- 
2.29.0

